pkgs/by-name/lm/lmstudio/darwin.nix

```diff
@@ -30,24 +29,5 @@ stdenv.mkDerivation {
   # LM Studio ships Scripts inside the App Bundle, which may be messed up by standard fixups
   dontFixup = true;
 
-  # undmg doesn't support APFS and 7zz does break the xattr. Took that approach from https://github.com/NixOS/nixpkgs/blob/a3c6ed7ad2649c1a55ffd94f7747e3176053b833/pkgs/by-name/in/insomnia/package.nix#L52
```

Comment:
@wegank We should probably avoid 7zz because of this.  Also we can't cross compile to darwin anyway so this shouldn't make things worse.

Reply:
Cross-compiling to Darwin is a stretch goal for 25.11. What is the issue with xattrs?

Reply:
I think @Prince213 meant https://github.com/NixOS/nixpkgs/pull/372738#issuecomment-2584758503, although that doesn't really justify switching to hdiutil in `airbuddy`, `aldente`, or `wechat` in #416709.

Reply:
> I think @Prince213 meant [#372738 (comment)](https://github.com/NixOS/nixpkgs/pull/372738#issuecomment-2584758503), although that doesn't really justify switching to hdiutil in `airbuddy`, `aldente`, or `wechat` in #416709.

Yes I'm worried about the xattrs issue.

I think it may be better if we have this in fetchzip as nativeBuildInputs.
If we allow networking for FODs it should also be okay to allow relaxed sandbox.

What do you think @reckenrode and @wegank?

Reply:
They both create FODs so I think the point is still valid.

Reply:
But with `fetchurl` I would probably call `/usr/bin/hdiutil` in a normal derivation, I guess?

Reply:
Doesn't feel right.  It's not really part of the build process, just part of the unpackPhase.

Reply:
The Nix store does not support xattrs so I don't think workarounds to keep them are a good idea. They will fundamentally break in cached NARs etc. That would need fixing on the Nix end.

Reply:
> What do you think @reckenrode and @wegank?

As @emilazy notes, the store doesn’t support xattrs. While a local build may be able to preserve them, a cached download from Hydra won’t have them.

The problem here is upstream apparently signed everything in their app bundle, including non-binaries. macOS supports that; the signatures are stored in xattrs. Unfortunately, the store support them. 🫤

Reply:
https://github.com/NixOS/nix/issues/185